src/hotspot/share/ci/ciObjectFactory.hpp

```diff
@@ -29,6 +29,8 @@
 #include "ci/ciObject.hpp"
 #include "utilities/growableArray.hpp"
 
+enum class vmSymbolID : int;
```

Comment:
Do you need this declaration to avoid #include "classfile/vmSymbols.hpp" ?

Reply:
Yes, that's the intention, because vmSymbols.hpp is big.

I am thinking whether I should move such forward declarations of enums into globalDefinitions.hpp, so header files that use `vmSymbolID` don't need to know that the base type is `int`.

Reply:
How about having a new header file utilities/vmEnums.hpp that looks like this:

```
// Include this header file if you just need the following enum types and
// you don't use their members directly. This way you don't need to include the
// complex header files that have the full definitions of these enums.
enum JVMFlagsEnum : int;
enum class vmSymbolID : int;
```

My plan is to have at least 2 more such enums (one for `vmIntrinsics::ID`, and one for `SystemDictionary::WKID`).

`: int` is the base type of the enum. I think that's part of the type's implementation so it shouldn't be exposed to the user of these types. If we litter `enum class vmSymbolID : int;` all over, when we decide to change the base type to something else, we would need to edit all such places.

Worse, if your forward declaration is wrong, like

```
enum class vmSymbolID : char;
```

the compiler won't detect this if your source file doesn't *also* include vmSymbols.hpp, so it could be dangerous.

Reply:
I am okay with this suggestion.

Reply:
I've moved the forward declarations into vmEnum.hpp in the latest version.

Reply:
Ok, I can't find it. Why did you have to add this little file to replace two forward declarations?  What was wrong with the forward declarations?